saliency_metrics/metrics/sanity_check/sanity_check.py

```diff
@@ -0,0 +1,72 @@
+from copy import deepcopy
+from typing import Dict, List, Optional, Union
+
+import numpy as np
+import torch
+from skimage.metrics import structural_similarity
+from torch import Tensor
+
+from saliency_metrics.metrics.build_metric import METRICS, ReInferenceMetric
+from saliency_metrics.metrics.sanity_check.sanity_check_result import SanityCheckResult
+from saliency_metrics.metrics.serializable_result import SerializableResult
+from saliency_metrics.models.model_utils import freeze_module
+
+
+class SanityCheck(ReInferenceMetric):
+    def __init__(self, ori_model: torch.nn.Module, perturb_layers: List[str], summarized: bool = True) -> None:
+        self._result = SanityCheckResult(summarized)
+        self.ori_model = ori_model  # Original Model
+        self._ori_state_dict = self.ori_model.state_dict()
+        self.model = deepcopy(self.ori_model)  # Model which will be used for performing randomization on
+        self.sd = self.model.state_dict()
+        self.perturb_layers = perturb_layers
+        self.all_layers: List = []
+        for i, x in self.ori_model.named_parameters():
+            self.all_layers.append(i)
+        self._classifier_layers = self._filter_names(self.all_layers)
+
+    def evaluate(self, image: Tensor, smap: np.ndarray, target: int, **kwargs) -> Dict:
+        ssim_list: List = []
+        for layer in self.perturb_layers:
+            self._reload_ckpt()
+            consecutive_perturb_layers: List = []
+            for j in self._classifier_layers[self._classifier_layers.index(layer) :]:
```

Comment:
Wrong. 1. The layers should be perturbed from the last to the first one. But `index` only gives you the index of the first seen `layer`. However, it is possible that a model has two layers with identical names. In this case, the implementation is wrong. 2. PLEASE DO NOT USE `j` in this case, because `j` is usually used to denote index. But here the element from the list should be a string. This is very ambiguous and will dramatically increase maintenance burden.